diesel_codegen/src/util.rs

```diff
@@ -84,6 +84,23 @@ pub fn is_option_ty(ty: &Ty) -> bool {
     }
 }
 
+pub fn inner_of_option_ty(ty: &Ty) -> Option<&Ty> {
+    use syn::PathParameters::AngleBracketed;
+
+    if !is_option_ty(ty) {
+        return None;
+    }
+
+    match *ty {
+        Ty::Path(_, Path { ref segments, .. }) =>
+            match segments[0].parameters {
+                AngleBracketed(ref data) => data.types.first(),
+                _ => None,
+            },
+        _ => None,
+    }
+}
```

Comment:
This basically extracts the `T` from `std::option::Option<T>`, right? (Might be interesting to see if syn can offer a simpler API here.)